src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -0,0 +1,175 @@
+
+    public ConversationService(UserRepository userRepository, CourseRepository courseRepository, AuthorizationCheckService authorizationCheckService,
+            ConversationParticipantRepository conversationParticipantRepository, ConversationRepository conversationRepository, SimpMessageSendingOperations messagingTemplate) {
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationRepository = conversationRepository;
+        this.messagingTemplate = messagingTemplate;
+    }
+
+    /**
+     * Persists given conversation
+     *
+     * @param courseId      id of course the conversation belongs to
+     * @param conversation  conversation to be persisted
+     * @return              persisted conversation
+     */
+    public Conversation createConversation(Long courseId, Conversation conversation) {
+        final User user = this.userRepository.getUserWithGroupsAndAuthorities();
+
+        if (conversation.getId() != null) {
+            throw new BadRequestAlertException("A new conversation cannot already have an ID", CONVERSATION_ENTITY_NAME, "idexists");
+        }
+
+        if (conversation.getConversationParticipants().isEmpty()) {
+            throw new BadRequestAlertException("A new conversation must have conversationParticipants", CONVERSATION_DETAILS_ENTITY_NAME, "NotNull");
+        }
+
+        final Course course = preCheckUserAndCourse(user, courseId);
+        conversation.setCourse(course);
+
+        Conversation savedConversation = conversationRepository.save(conversation);
+
+        ConversationParticipant conversationParticipantOfCurrentUser = new ConversationParticipant();
+        conversationParticipantOfCurrentUser.setUser(user);
+        conversation.getConversationParticipants().add(conversationParticipantOfCurrentUser);
+
+        conversation.getConversationParticipants()
+                .forEach(conversationParticipant -> conversationParticipant = createConversationParticipant(conversationParticipant, savedConversation));
+
+        // informs involved users about a new conversation
+        broadcastForConversation(new ConversationDTO(savedConversation, CrudAction.CREATE));
+
+        return savedConversation;
+    }
+
+    /**
+     * fetch conversation from database by conversationId
+     *
+     * @param conversationId  id of the conversation to fetch
+     * @return  fetched conversation
+     */
+    public Conversation getConversationById(Long conversationId) {
+        Optional<Conversation> conversation = conversationRepository.findById(conversationId);
+        conversation.ifPresent(conversation1 -> conversation1
+                .setConversationParticipants(new HashSet<>(conversationParticipantRepository.findConversationParticipantByConversationId(conversationId))));
```

Comment:
Please reduce this into a single database call by using an eager fetch method in the ConversationRepository.

Reply:
I initially had an eager fetch but stephan asked me to fetch it lazily. This way we don't fetch all conversation details while fetching posts from the database.

Reply:
Please take a look at some existing repositories like `ProgrammingExerciseRepository`. You can fetch eagerly when needed by using `LEFT JOIN FETCH` in you database queries.

Reply:
Done